### PR TITLE
[serverless-apollo-contentful] Add queueing to kit

### LINKED ISSUE #863: [serverless-apollo-contentful] Add queueing to kit
# Description

As applications grow, the need for asynchronous workers become needed. Newer developers are less familiar with this pattern. As such, we want to include this as part of the starter kit as an opt-in feature so it can be easily used by developers.

SQS is the answer (see how Dustin did it in his kit, should be roughly identical). It's not the event processor for DynamoDB but the endpoint that generated an example job. There's a use case here where you get a hook back from contentful on a model change and need to do some processing in the background. Set it up as something basic, but document in the README about it.

Image

### PR DESCRIPTION
**Pull request by kodejuice**
## Type of change

- [x] Feature
- [ ] Documentation change
- [ ] Bug fix

## Summary of change

- [ ] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [ ] This starter kit has been approved by the maintainers
- [ ] Changes for this new starter kit are being pushed to an integration branch instead of main
- [ ] All dependencies are version locked
- [x] This fix resolves #863 
- [ ] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/graphql-serverless-contentful/package.json [hunk 1]
```diff
@@ -7,12 +7,15 @@
...
+		"infrastructure:remove": "docker compose down",
+		"infrastructure:start": "docker compose start",
+		"infrastructure:stop": "docker compose stop",
 		"deploy": "sls deploy",
 		"format": "prettier --write \"**/*.{ts,js,json,md}\"",
 		"ts:check": "tsc --noEmit",
 		"codegen": "graphql-codegen --config codegen.ts",
-		"db:seed": "ts-node ./scripts/seed.ts",
-		"db:migration": "ts-node ./scripts/migration.ts"
+		"seed": "ts-node ./scripts/seed.ts"
```
Comment (kodejuice) [inline starters/graphql-serverless-contentful/package.json:18]:
```suggestion
                "db:seed": "ts-node ./scripts/seed.ts",
		"db:migration": "ts-node ./scripts/migration.ts"
```

File: starters/graphql-serverless-contentful/src/handlers/sqs-handler.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { SQSHandler } from 'aws-lambda';
+
+export const handler: SQSHandler = async (event) => {
+	const recordHandler = async (record: any) => {
+		console.log(JSON.stringify(record.body));
```
[the remark this answers is not in the document]
Comment (mbicknese) [reply starters/graphql-serverless-contentful/src/handlers/sqs-handler.ts:5]: Interesting talk to have on what we want to demo. For now, this is implemented as specified, and we should keep it.

File: starters/graphql-serverless-contentful/src/utils/sqs/client.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+let cachedClient: SQSClient | null = null;
+
+export const getClient = (): SQSClient => {
+	if (cachedClient) {
+		return cachedClient;
+	}
+
+	const config: SQSClientConfig = {};
+	if (isOffline()) {
+		config.endpoint = 'http://localhost:9324';
```
[the remark this answers is not in the document]
Comment (kodejuice) [reply starters/graphql-serverless-contentful/src/utils/sqs/client.ts:13]: Not really, this is offline so will always be `localhost`

File: starters/graphql-serverless-contentful/src/handlers/sqs-handler.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kodejuice) [reply starters/graphql-serverless-contentful/src/handlers/sqs-handler.ts:5]: I think its ok as is 

image

File: starters/graphql-serverless-contentful/package.json [hunk 4]
```diff
@@ -25,10 +29,13 @@
 	"dependencies": {
 		"@apollo/server": "4.3.0",
 		"@as-integrations/aws-lambda": "1.1.0",
+		"@aws-sdk/client-sqs": "^3.259.0",
+		"aws-sdk": "^2.1306.0",
```
Comment (iansamz) [inline starters/graphql-serverless-contentful/package.json:33]: Please lock dependecies

**Pull request merged**